pkg/namespacestore/validation.go

```diff
@@ -28,6 +28,11 @@ func ValidateNamespaceStore(nsStore *nbv1.NamespaceStore) error {
 		return ValidateNsStoreNSFS(nsStore)
 
 	case nbv1.NSStoreTypeAWSS3:
+		if nsStore.Spec.AWSS3.AWSSTSRoleARN != nil {
```

Comment:
> I think it will be best to separate these validation to different functions, to be more accurate with error messages and also to keep function responsibility clear and single as a good design.

also here.

Reply:
Done